include/seqan3/alignment/pairwise/alignment_configurator.hpp

```diff
@@ -33,6 +33,36 @@
 
 namespace seqan3::detail
 {
+//!\cond
+template <typename value_t>
+SEQAN3_CONCEPT align_pairwise_value_concept =
+    tuple_like_concept<value_t> &&
+    std::tuple_size_v<value_t> == 2 &&
+    std::ranges::ForwardRange<std::tuple_element_t<0, value_t>> &&
+    std::ranges::ForwardRange<std::tuple_element_t<1, value_t>>;
+
+template <typename value_t>
+SEQAN3_CONCEPT align_pairwise_single_input_concept =
+    align_pairwise_value_concept<value_t> &&
+    std::ranges::ViewableRange<std::tuple_element_t<0, value_t>> &&
+    std::ranges::ViewableRange<std::tuple_element_t<1, value_t>>;
+
+// Only use input ranges whose value type models align_pairwise_value_concept and
+// whose reference type is an lvalue reference and the range itself models ViewableRange or
+// the reference type is a prvalue and it models align_pairwise_single_input_concept.
+// This covers all typical use cases:
+// a) A lvalue range, whose reference type is a tuple like lvalue reference,
+// b) A range, whose reference type is a tuple over viewable ranges.
+// This covers also transforming and non-transforming views (e.g. std::ranges::view::zip, or view::take).
+// Only a temporary non-view range piped with view::persist can't be handled securely.
+template <typename range_t>
+SEQAN3_CONCEPT align_pairwise_range_input_concept =
+    std::ranges::InputRange<std::remove_reference_t<range_t>> &&
+    align_pairwise_value_concept<value_type_t<std::remove_reference_t<range_t>>> &&
+    ((std::ranges::ViewableRange<range_t> && std::is_lvalue_reference_v<reference_t<range_t>>) ||
+     align_pairwise_single_input_concept<std::remove_reference_t<reference_t<range_t>>>);
+
+//!\endcond
```

Comment:
why is this \cond? Please add at least short developer documentation.
Also concepts are CamelCase now.